.gitignore

```diff
@@ -37,3 +37,9 @@
 # -- --
 
 .vscode
+
+# -- --
+# Python
+# -- --
+
+venv
```

Comment:
What's the motivation for this advice? I don't see any risk/challenge with having a slightly too big gitignore. Is there something I am missing?

I created the venv to work well with VSCode. There is probably a different manner for me to configure it? In any case venv is Python specific so it is very relevant for our repo, right?

Reply:
The previous discussion was here: https://github.com/tournesol-app/tournesol/pull/699.

The main problem with adding user specific rules into the project's `.gitignore` is just messiness and maintenance, I think. The project is cleaner and easier to maintain without all the rules contributors may want. It's much easier to decide what these files should contain and what can be removed if the plethora of possible user environments is excluded.

Using a global gitignore is also helpful for the contributors themselves because they don't have to add their own rules to every project they work on. If you use VSCode then add `.vscode` to it and it's done for all your projects, including future ones. If you always name your venv directories `venv` then add it to your global gitignore and you're done.

I think the rule should be to only add files that the project generates when it's used. This includes automatically generated files (`*.pyc`, `node_modules/`, `dist/`, etc.) but also files and directories that the documentation recommends to create. For example, I would only add `venv` if a documentation suggests to create this particular directory. Otherwise why wouldn't you also add `.venv`, `myenv`, `venv2` or all the possible names people might use for their venvs?

To configure your global gitignore you can follow [this small blog post](https://sebastiandedeyne.com/setting-up-a-global-gitignore-file/) that also provides some motivations.

Reply:
Thanks for the detailed explaination 👍